OpenTelemetry integration page (#3585)
neondatabase/website: content/docs/guides/opentelemetry.md

---
title: OpenTelemetry integration
subtitle: Send Neon metrics and Postgres logs to any OTEL-compatible observability platform
enableTableOfContents: true
updatedOn: '2025-01-16T10:00:00.000Z'
---

<FeatureBetaProps feature_name="OpenTelemetry integration" />

<InfoBlock>
<DocsList title="What you will learn:">
<p>How to configure OpenTelemetry exports from Neon</p>
<p>Example config using New Relic</p>
</DocsList>

<DocsList title="External docs" theme="docs">
<a href="https://opentelemetry.io/docs/specs/otlp/">OpenTelemetry Protocol (OTLP) Specification</a>
<a href="https://docs.newrelic.com/docs/opentelemetry/best-practices/opentelemetry-otlp/">New Relic OpenTelemetry guide</a>
</DocsList>
</InfoBlock>

Available for Scale and Business Plan users, the Neon OpenTelemetry integration lets you export metrics and Postgres logs to any OpenTelemetry Protocol (OTLP) compatible observability platform. This gives you the flexibility to send your Neon data to your preferred monitoring solution, whether that's New Relic, Grafana Cloud, Honeycomb, or any other OTEL-compatible service.

> If you don't already have an OTEL-compatible platform set up, we'll walk you through the basic setup in New Relic so you can try out how Neon's data export works. The same configuration principles apply to any OTEL-compatible platform.

Reviewer: ```suggestion
If you don't already have an OTEL-compatible platform set up, we'll walk you through the basic setup in New Relic so you can see how Neon's data export works. The same configuration principles apply to any OTEL-compatible platform.
```

Choose what data you want to export (at the top of the form):

- **Metrics**: System metrics and database statistics (CPU, memory, connections, etc.)
- **Postgres logs**: Error messages, warnings, connection events, and system notifications

You can enable either or both options based on your monitoring needs.

## Configure the connection

1. Select your connection protocol. For most platforms including New Relic, choose **HTTP** (recommended), which uses HTTP/2 for efficient data transmission. Some environments may require **gRPC** instead.

2. Enter your **Endpoint** URL.

   For New Relic, enter:

   - US: `https://otlp.nr-data.net`
   - Europe: `https://otlp.eu01.nr-data.net`

   See [this table](https://docs.newrelic.com/docs/opentelemetry/best-practices/opentelemetry-otlp/#configure-endpoint-port-protocol) for other options.

3. Configure authentication:

   - **Bearer**: Enter your bearer token or API key
   - **Basic**: Provide your username and password credentials
   - **API Key**: Enter your API key

   For New Relic, you can use either **Bearer** or **API Key** authentication with your New Relic license key (both work the same way).

## Configure resource attributes

> Neon automatically organizes your data into separate service entities: your configured service name will receive PostgreSQL logs, while metrics are split into `compute-host-metrics` (infrastructure metrics) and `sql-metrics` (database metrics).

Reviewer: ```suggestion
Neon automatically organizes your data into separate service entities: your configured service name will receive Postgres logs, while metrics are split into `compute-host-metrics` (infrastructure metrics) and `sql-metrics` (database metrics).
```

1. In the **Resource** section, configure the `service.name` attribute to identify your Neon project in your observability platform. For example, you might use "neon-postgres-test" or your actual project name.

2. Optionally, you can add additional resource attributes by providing a value in the second field to further categorize or filter your data in your observability platform.

## Complete the setup

Click **Add** to save your configuration and start the data export.

## Verify your integration is working

Your Neon data should start appearing in your observability platform within a few minutes.

**For New Relic users**, use these queries to check if data is flowing:

```sql
FROM Metric SELECT * SINCE 1 hour ago
FROM Log SELECT * SINCE 1 hour ago
```

**Success looks like this:**

Metrics flowing into New Relic
![Neon metrics appearing in New Relic](/docs/guides/new_relic_metrics_success.png)

> PostgreSQL logs

Reviewer: ```suggestion
Postgres logs
```
